torch_geometric/nn/models/explainer.py

```diff
@@ -144,3 +149,211 @@ def to_captum(model: torch.nn.Module, mask_type: str = "edge",
+                 return_type: str = 'log_prob', log: bool = False):
+        super().__init__()
+        assert return_type in ['log_prob', 'prob', 'raw', 'regression']
+
+        self.model = model
+        self.lr = lr
+        self.epochs = epochs
+        self.num_hops = num_hops
+        self.return_type = return_type
+        self.log = log
+
+    @property
+    def num_hops(self):
+        return self._num_hops
+
+    @num_hops.setter
+    def num_hops(self, num_hops):
+        if num_hops is not None:
+            self._num_hops = num_hops
+        else:
+            # Could this be a general utils function?
+            k = 0
+            for module in self.model.modules():
+                if isinstance(module, MessagePassing):
+                    k += 1
+            self._num_hops = k
+
+    def _flow(self):
+        for module in self.model.modules():
+            if isinstance(module, MessagePassing):
+                return module.flow
+        return 'source_to_target'
+
+    def subgraph(self, node_idx, x, edge_index, **kwargs):
+        r"""Returns the subgraph of the given node.
+        Args:
+            node_idx (int): The node idx to explain.
+            x (Tensor): The node feature matrix.
+            edge_index (LongTensor): The edge indices.
+            **kwargs (optional): Additional arguments passed to the GNN module.
+        :rtype: (Tensor, Tensor, LongTensor, LongTensor, LongTensor, dict)
+        """
+        num_nodes, num_edges = x.size(0), edge_index.size(0)
+        subset, edge_index, mapping, edge_mask = k_hop_subgraph(
+            node_idx, self.num_hops, edge_index, relabel_nodes=True,
+            num_nodes=num_nodes, flow=self._flow())
+
+        x = x[subset]
+        kwargs_new = {}
+        for key, value in kwargs.items():
+            if torch.is_tensor(value) and value.size(0) == num_nodes:
+                kwargs_new[key] = value[subset]
+            elif torch.is_tensor(value) and value.size(0) == num_edges:
+                kwargs_new[key] = value[edge_mask]
+            kwargs_new[key] = value  # TODO: this is not in PGExplainer
+        return x, edge_index, mapping, edge_mask, subset, kwargs_new
+
+    def _to_log_prob(self, x):
+        x = x.log_softmax(dim=-1) if self.return_type == 'raw' else x
+        x = x.log() if self.return_type == 'prob' else x
+        return x
+
+    @torch.no_grad()
+    def get_initial_prediction(self, x, edge_index, batch=None, **kwargs):
+        if batch is not None:
+            out = self.model(x, edge_index, batch, **kwargs)
+        else:
+            out = self.model(x, edge_index, **kwargs)
+        if self.return_type == 'regression':
+            prediction = out
+        else:
+            log_logits = self._to_log_prob(out)
+            prediction = log_logits.argmax(dim=-1)
+        return prediction
+
+    def get_loss(self, out, prediction, node_idx=None, **kwargs):
```

Comment:
Is `node_idx` better `-1` or `None` in graph classification cases?

Reply:
I prefer `None`.